File: lib/domain/crop/dtos/add_crop_crop_snapshot.rb
# frozen_string_literal: true

module Domain
  module Crop
    module Dtos
      # REST add_crop: 作物解決サブステップの戻り値（plan_crop 作成・候補探索で参照するフィールドのみ）。
      class AddCropCropSnapshot
        attr_reader :id, :name, :variety, :area_per_unit, :revenue_per_area

        def initialize(id:, name:, area_per_unit:, revenue_per_area:, variety: nil)
          @id = id
          @name = name
          @variety = variety
          @area_per_unit = area_per_unit
          @revenue_per_area = revenue_per_area
          freeze
        end
      end
    end
  end
end

File: crates/agrr-domain/src/crop/dtos/add_crop_crop_snapshot.rs
//! Ruby: `Domain::Crop::Dtos::AddCropCropSnapshot`

#[derive(Debug, Clone, PartialEq)]
pub struct AddCropCropSnapshot {
    pub id: i64,
    pub name: String,
    pub variety: Option<String>,
    pub area_per_unit: f64,
    pub revenue_per_area: f64,
}
